src/Migrations/M031MigrateMediaBlockToAudioVideoBlock.php

```diff
@@ -0,0 +1,365 @@
+        }
+    }
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Get the media URL from a Media block.
+     * If the value is not a URL then it is a Youtube ID.
+     * In that case, the ID is turned into a full Youtube URL.
+     *
+     * @param array $block - A parsed Media block.
+     * @return string - The media URL or null if no media was set.
+     */
+    private static function get_media_url(array $block): mixed
+    {
+        $media_url = $block['attrs']['youtube_id'] ?? $block['attrs']['media_url'] ?? null;
+
+        // Replace special characters in the url.
+        $media_url = strtr($media_url, '&', 'u0026');
+
+        if ($media_url && !filter_var($media_url, FILTER_VALIDATE_URL)) {
+            $media_url = 'https://www.youtube.com/watch?v=' . $media_url;
+        }
+
+        return $media_url;
+    }
+
+    /**
+     * Get the source of a Media block.
+     * It can be Youtube, Vimeo, Souncloud, or an .mp3, .mp4 or .wav file.
+     *
+     * @param string $url - The Media URL used to identify the source.
+     * @return string - The source of the media block or an empty string if no source was found.
+     */
+    private static function identify_source(string $url): mixed
+    {
+        foreach (self::SOURCE_PATTERNS as $value => $key) {
+            if (preg_match($key, $url)) {
+                return $value;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Transform a Media block into a group of blocks.
+     * This group contains a header for the media block title,
+     * a paragraph for the media block description,
+     * and a final block (embed, video or audio) for the media source.
+     *
+     * @param array $block - A parsed Media block.
+     * @param string $source - The source of the Media block.
+     * @param string $media_url - The Media URL.
+     * @return array - The transformed block.
+     */
+    private static function transform_blocks(array $block, string $source, string $media_url): array
+    {
+        $block_title = array_key_exists("video_title", $block['attrs']) ? $block['attrs']['video_title'] : null;
+        $block_description = array_key_exists("description", $block['attrs']) ? $block['attrs']['description'] : null;
+
+        $block = [];
+        $block['blockName'] = Utils\Constants::BLOCK_GROUP;
+        $block['attrs']['metadata']['name'] = 'Media Group';
+
+        $block['innerBlocks'] = [];
+        $block['innerBlocks'][0] = $block_title ? self::get_heading_block($block_title) : null;
+        $block['innerBlocks'][1] = $block_description ? self::get_paragraph_block($block_description) : null;
+        $block['innerBlocks'][3] = self::get_media_block($source, $block, $media_url);
+
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
+        $block['innerHTML'] =
+        '<div class="wp-block-group">
+
+
+
+
+
+        </div>';
+
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
```

Comment:
I tried a different approach to transform a block in a [split 2 columns block](https://github.com/greenpeace/planet4-master-theme/pull/2350/files#diff-caf19d90fc201d2db867d04cd8f157f6bb37183e6113012af080bec25d252c5cR95-R139) migration, we may use similar solution here as well.

Reply:
Thanks @sagarsdeshmukh 
I have a slightly different opinion about the approach.
Please, check my comment [here](https://github.com/greenpeace/planet4-master-theme/pull/2350#discussion_r1751540000), and let's discuss it further.